model/src/main/java/de/danoeh/antennapod/model/feed/FeedPreferences.java

```diff
@@ -168,12 +191,30 @@ public void setFeedID(long feedID) {
         this.feedID = feedID;
     }
 
-    public boolean getAutoDownload() {
-        return autoDownload;
+    public boolean getAutoDownload(boolean globalDefault) {
+        switch (this.autoDownload) {
+            case ENABLED -> {
+                return true;
+            }
+            case DISABLED -> {
+                return false;
+            }
+            default -> {
+                return globalDefault;
+            }
```

Comment:
Hmm, this needs more lines than the "normal" switch syntax. Not sure if we should do this.

Reply:
What do you mean with "normal" switch syntax? Do you mean two if's and then a return similar to line 290 in that file? Also, does it matter how "long" this switch is, if it is easily understandable?

Reply:
Originally, switches in Java looked like this:
```
switch(...) {
    case ENABLED:
        return false;
```

Java 12 introduced switch expressions:

```
return switch(...) {
    case ENABLED -> false;
```

What is used here is some kind of mix between a switch expression (with the arrow `->`) and the statement (with colon and return). Everywhere else in the app, we use the "original" switch syntax.

Reply:
Oh yeah. I am switching so often between java and kotlin, i just use what the ide suggests :D

I updated it to switch expressions because it fits good here, but i can change it back to the original one if you like.